aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -492,8 +486,117 @@ impl AggregatorData {
         Ok(snapshot_id)
     }
 
-    pub fn read_snapshot(&self, _id: AggregatorVersionedID) -> PartialVMResult<u128> {
-        unimplemented!();
+    pub fn create_new_snapshot(&mut self, id: AggregatorID, value: SnapshotValue) {
+        let snapshot_state = AggregatorSnapshotState::Create { value };
+
+        self.aggregator_snapshots.insert(id, AggregatorSnapshot {
+            id,
+            state: snapshot_state,
+        });
+    }
+
+    /// Returns a reference to an AggregatorSnapshot with `id`.
+    /// If transaction that is currently executing did not initialize it,
+    /// a new AggregatorSnapshot instance is created.
+    /// Note: when we say "aggregator snapshot instance" here we refer to Rust struct and
+    /// not to the Move aggregator snapshot.
+    pub fn get_snapshot(
+        &mut self,
+        id: AggregatorID,
+        resolver: &dyn AggregatorResolver,
+    ) -> PartialVMResult<&AggregatorSnapshot> {
+        let snapshot = match self.aggregator_snapshots.entry(id) {
+            Entry::Vacant(entry) => {
```

Comment:
In this case, why not simply call `snapshot` function, and return its output?
Why are we doing an expensive read to create a snapshot here?

Reply:
I think it was confusing to have this as a separate function, this is code for read_snapshot, so moved it there.